ctapipe/image/cleaning.py

```diff
@@ -440,3 +575,34 @@ def __call__(
             min_number_neighbors=self.min_picture_neighbors.tel[tel_id],
             time_limit=self.time_limit_ns.tel[tel_id],
         )
+
+
+class TimeConstrainedImageCleaner(TailcutsImageCleaner):
+    """
+    MAGIC-like Image cleaner with timing information (See `ctapipe.image.time_constrained_clean`)
+    """
+
+    time_limit_core_ns = FloatTelescopeParameter(
+        default_value=4.5, help="arrival time limit for neighboring " "core pixels, in ns"
+    ).tag(config=True)
+    time_limit_boundary_ns = FloatTelescopeParameter(
+        default_value=1.5, help="arrival time limit for neighboring " "boundary pixels, in ns"
+    ).tag(config=True)
+
+    def __call__(
+        self, tel_id: int, image: np.ndarray, arrival_times=None
+    ) -> np.ndarray:
+        """
+        Apply MAGIC-like image cleaning with timing information. See `ImageCleaner.__call__()`
+        """
+
+        return mars_cleaning_1st_pass(
```

Comment:
Do you really want to run this function here, and not your `time_constrained_clean()` function?

Reply:
No, this I should have changed to `time_constrained_clean()`. I will do it now...